api/app/Http/Controllers/AuthController.php

```diff
@@ -212,7 +217,52 @@ public function handleProviderCallback($provider, Socialite $socialite, UserRepo
     }
 
     /**
-     * Check so the provider exists.
+     * Provide a requester with user information for single sign on.
+     *
+     * @param  string  $requester
+     * @param  Request $request
+     *
+     * @return Response
+     */
+    public function singleSignOn($requester, Request $request)
+    {
+        if ($token = $request->cookie('JwtAuthToken')) {
+            $user = $this->jwtAuth->toUser($token);
+        } else {
+            $user = null;
+        }
+
+        $requester = SingleSignOnFactory::create($requester, $request, $user);
+
+        return $requester->getResponse();
+    }
+
+
+    /**
+     * Prepare the token to be passed as cookie.
+     *
+     * @param  string  $token
+     * @param  Request $request
+     *
+     * @return Cookie
+     */
+    protected function tokenToCookie($token, Request $request)
+    {
+        // Get the domain
+        $segments = explode('.', $request->getHost());
+        if (count($segments) > 1) {
+            $segments = array_slice($segments, -2);
+        }
+        $domain = implode('.', $segments);
```

Comment:
This implementation feels somewhat unstable, I think it would be better to have a configurable cookie domain in the dotfiles (make it defined in the `.common.env` file via laravel)